server/services/DeckService.js

```diff
@@ -440,6 +441,17 @@ class DeckService {
             }
 
             decksByUuid[dbDeck.uuid] = dbDeck;
+            expansionDeck.push(dbDeck.expansion);
```

Comment:
Just store the expansionid the first time round and then compare against it, returning error if it doesn't match. No need to do all this array comparing

Reply:
fixed